### PR TITLE
README: Update README structure and description

### PR DESCRIPTION
**Pull request by dghubble**
* Updates matchbox description cc @philips 
* Links to coreos docs and structures the README to be similar to those docs cc @ElijahCaine

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,19 +4,11 @@ Network boot and provision Container Linux clusters on virtual or physical hardw
...
+`matchbox` is a service that matches machines (based on labels like MAC, UUID, etc.) to profiles to PXE boot and provision Container Linux clusters. Profiles specify the kernel/initrd, kernel arguments, iPXE config, GRUB config, [Container Linux Config](Documentation/container-linux-config.md), [Cloud-Config](Documentation/cloud-config.md), or other configs a machine should use. Matchbox can be [installed](Documentation/deployment.md) as a binary, RPM, container image, or deployed on a Kubernetes cluster and it provides an authenticated gRPC API for clients like [terraform](https://github.com/coreos/terraform-provider-matchbox).
```
Comment (ElijahCaine) [inline README.md]: Please put all links (references) at the bottom of the file:
```markdown
... [Container Linux config][cl-config] ...
...
[cl-config]: Documentation/container-linux-config.md
```

File: README.md [hunk 2]
```diff
@@ -4,19 +4,11 @@ Network boot and provision Container Linux clusters on virtual or physical hardw
...
 
+* [Documentation](https://coreos.com/matchbox/docs/latest)
```
Comment (ElijahCaine) [inline README.md]: See the links at the bottom request.

File: README.md (hunk 1 above)
Comment (dghubble) [inline README.md]: I can for this paragraph. The page itself is almost entirely links so I dunno if it makes sense for the bullet points.

**Comment by dghubble** [comment]
I've used that style for links that were duplicated or that you mentioned. I hesitate to change them all because it seems to make refactors harder - it'd be easy to accumulate dead links at the bottom and you need a parser to check if they're unused, etc.

**Pull request merged**